app/src/main/java/com/google/android/sambadocumentsprovider/auth/AuthActivity.java

```diff
@@ -0,0 +1,165 @@
+import com.google.android.sambadocumentsprovider.document.DocumentMetadata;
+import com.google.android.sambadocumentsprovider.nativefacade.SmbClient;
+
+import java.io.IOException;
+
+public class AuthActivity extends AppCompatActivity {
+  private static final String TAG = "AuthActivity";
+
+  public static final String SHARE_URI_KEY = "shareUri";
+
+  private SmbClient mClient;
+  private ShareManager mShareManager;
+  private DocumentMetadata mMetadata;
+  private ShareManager.ShareMountChecker mChecker;
+
+  private EditText nameText;
+  private EditText passwordText;
+  private EditText domainText;
+  private CheckBox mountCheckBox;
+  private ProgressBar progressBar;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_auth);
+
+    Intent intent = getIntent();
+    Uri shareUri = Uri.parse(intent.getStringExtra(SHARE_URI_KEY));
+
+    mMetadata = DocumentMetadata.createShare(shareUri);
+
+    mClient = SambaProviderApplication.getSambaClient(getApplicationContext());
+    mShareManager = SambaProviderApplication.getServerManager(getApplicationContext());
+
+    TextView shareText = (TextView) findViewById(R.id.shareTextView);
+    nameText = (EditText) findViewById(R.id.editName);
+    passwordText = (EditText) findViewById(R.id.editPassword);
+    domainText = (EditText) findViewById(R.id.editDomain);
+    mountCheckBox = (CheckBox) findViewById(R.id.checkBox);
+    progressBar = (ProgressBar) findViewById(R.id.progressBar);
+
+    shareText.setText(shareUri.toString());
+
+    mChecker = new ShareManager.ShareMountChecker() {
+      @Override
+      public void checkShareMounting() throws IOException {
+        mMetadata.loadChildren(mClient);
+      }
+    };
+  }
+
+  public void onLoginClick(View view) {
+    progressBar.setVisibility(View.VISIBLE);
+
+    if (TextUtils.isEmpty(nameText.getText()) || TextUtils.isEmpty(passwordText.getText())) {
+      Toast.makeText(this, "Username and password can't be empty!", Toast.LENGTH_SHORT).show();
+      return;
+    }
+
+    new AuthorizationTask(
+      nameText.getText().toString(),
+      passwordText.getText().toString(),
+      domainText.getText().toString(),
+      mChecker).execute();
+  }
+
+  private void onLoginFinished(boolean success) {
+    setResult(RESULT_OK);
+    finish();
+  }
+
+  private class AuthorizationTask extends BiResultTask<Void, Void, Void> {
+    private final String mUser;
+    private final String mPassword;
+    private final String mDomain;
+
+    private final ShareManager.ShareMountChecker mChecker;
+
+    AuthorizationTask(
+            String user, String password, String domain, ShareManager.ShareMountChecker checker) {
+      mUser = user;
+      mPassword = password;
+      mDomain = domain;
+
+      mChecker = checker;
+    }
+
+    @Override
+    public Void run(Void... voids) throws Exception {
+      if (mountCheckBox.isChecked()) {
```

Comment:
Yes, I realized that when I did it. I mainly wanted to give user the choice of not flooding his left pane with shares. Another option would be to implement logic to store the credentials permanently but not pin the share. Or put a warning that if the box is not checked, credentials will be needed again after rerun. What do you think?